src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -112,48 +132,59 @@ public override IList<ITestResult> Children
         /// <param name="result">The result to be added</param>
         public virtual void AddResult(ITestResult result)
         {
-            Children.Add(result);
+#if PARALLEL
+            _children.Enqueue(result);
+#else
+            _children.Add(result);
+#endif
 
```

Comment:
Not sure what to do about this, since it breaks the virtualization of Children.  Can't use Children anymore, because it is an IEnumerable, which has no Add or equivalent methods.